lighthouse-core/lib/network-recorder.js

```diff
@@ -39,11 +45,82 @@ class NetworkRecorder extends EventEmitter {
   }
 
   isIdle() {
-    return this.activeRequestCount() === 0;
+    const quietPeriods = NetworkRecorder.findNetworkQuietPeriods(this._records, 0);
+    return quietPeriods.some(period => period.ongoing);
   }
 
   is2Idle() {
-    return this.activeRequestCount() <= 2;
+    const quietPeriods = NetworkRecorder.findNetworkQuietPeriods(this._records, 2);
+    return quietPeriods.some(period => period.ongoing);
+  }
+
+  _emitNetworkStatus() {
+    if (this.isIdle()) {
+      this.emit('networkidle');
+    } else {
+      this.emit('networkbusy');
+    }
+
+    if (this.is2Idle()) {
+      this.emit('network-2-idle');
+    } else {
+      this.emit('network-2-busy');
+    }
+  }
+
+  /**
+   * Finds all time periods where the number of inflight requests is less than or equal to the
+   * number of allowed concurrent requests.
+   * @param {!Array<!WebInspector.NetworkRequest>} networkRecords
+   * @param {number} allowedConcurrentRequests
+   * @param {number=} endTime
+   * @return {!Array<{start: number, end: number, ongoing: boolean|undefined}>}
```

Comment:
naming bikeshed! is ongoing === "in flight"? if so i'd prefer that as a name, but i suspect there's a difference already..... :)

Reply:
yeah they're different haha :)

it's a flag to signal that this period had no end, basically that it's `Infinity` even though we mark it at `endTime`, suggestions welcome :)